### DISCUSSION TITLE
RDF Graph Kernels

### DISCUSSION THREAD

**Discussion by GillesVandewiele** [RDFLib/rdflib#1598] [Development]
Would it be interesting to create an extra package that calculates different kernel metrics in a given RDF graph?

As in: http://www.ecmlpkdd2013.org/wp-content/uploads/2013/07/464.pdf (which I currently implemented (not the most efficient implementation though))

**Comment by joernhees** [comment]
definitely 👍, this has been on my list for quite some time now...

depending on the requirements, how big this gets and how critical "timely releases" are for you, i'd suggest to either create a graph kernels repo (if you want we can put it in the https://github.com/RDFLib namespace) or if you want to start small just make a PR in some new module (e.g., `kernel.py`)...

**Comment by GillesVandewiele** [comment]
Alright, my current implementation is 250 lines of code and has no dependencies except for rdflib (obviously) and copy. It also has two custom-made classes (one for a Graph and one for a Vertex).

I can do a PR, maybe I'll upload two files, one with the code and one with a working example?

**Comment by GillesVandewiele** [comment]
PR sent. #796
